python/cuml/dask/naive_bayes/naive_bayes.py

```diff
@@ -0,0 +1,205 @@
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import cupy as cp
+
+
+from uuid import uuid1
+
+from cuml.naive_bayes import MultinomialNB as MNB
+
+import dask
+
+from cuml.dask.common import extract_arr_partitions, \
+    workers_to_parts
+
+from dask.distributed import default_client
+
+
+class MultinomialNB(object):
+
+    """
+    Distributed Naive Bayes classifier for multinomial models
+
+    The multinomial Naive Bayes classifier is suitable for classification
+    with discrete features (e.g., word counts for text classification).
+
+    The multinomial distribution normally requires integer feature counts.
+    However, in practice, fractional counts such as tf-idf may also work.
```

Comment:
I'd like to add an example in this PR if I can, though I just realized one of the reasons I haven't added the example yet is because of the hacky way I needed to create the sparse cupy-backed dask.array from the scipy sparse array using the 20newgroups dataset. 

We have a few options (and possibly more):
1. Not add an example until we can reliably create a sparse cupy-backed Dask.Array from a cupy.sparse array.
2. Add a utility for now that converts a cpu-based scipy.sparse array to a cupy-backed sparse Dask.Array. 
3. Wait until Dask and CuPY properly support the needed scipy.sparse features to use Dask.array. 

#3 could take awhile so I think that might be out of the question. I hesitate to do #1 as I'm concerned it might mean doing #3. I could do #2 in the meantime. It's unfortunate, but it might be the most reasonable solution for now, given the first-class support is still pending. What do you think?

Reply:
I think you're correct, solution 2 seems to offer the best balance and it might be pretty useful while 3 happens entirely. It also "shields" us from bugs that could be introduced in the new cupy sparse features. Do you have a feel of what could be the expected timeline of the features?